js/README.md

```diff
@@ -1,111 +1,121 @@
-## Run this with different Android device IDs
-$ ANDROID_DEVICE=__ANDROID_DEVICE_ID_1__ make android.debug
-$ ANDROID_DEVICE=__ANDROID_DEVICE_ID_2__ make android.debug
+# Optional if using only one device
+# Run this with different Android device IDs
+$ ANDROID_DEVICE="your-emulator-id" make android.debug
 
-💡 You can check available Android Virtual Devices with `adb devices`
+# 💡 You can check available Android Virtual Devices with
+$ adb devices
+```
+
+```bash
+# Optional if not modifying any .proto file
+# Generate files using protobuf
+$ make generate
+
+# Optional if using only one device
+# Run this with different ports to test with multiple devices
+$ BERTY_DAEMON_PORT=1337 make daemon.start
+$ BERTY_DAEMON_PORT=1338 make daemon.start
 
-## Web
-## Run Berty daemon for the web app
+# Web
+# Run Berty daemon for the web app
 make web.daemon.start
-## Run web app in debug mode
+# Run web app in debug mode
 $ make web.debug
-## Open http://localhost:3002/#/ip4/127.0.0.1/tcp/9092/grpcws in your browser
+# Open http://localhost:3002/#/ip4/127.0.0.1/tcp/9092/grpcws in your browser
+```
+
+List of all available commands:
+
+```bash
+$ make help
 ```
 
-## 🧳 Requirements
+## Setup
 
-### Requirements for working on iOS and Android apps
+### The Berty Messenger installation involves:
 
-- The [General React Native requirements](#general-react-native-requirements)
-- [iOS dev reqs](#ios-dev-requirements) **and/or** [Android dev reqs](#android-dev-requirements)
-- [Watchman](https://facebook.github.io/watchman/docs/install/) to enable live reloading
-- [Docker Desktop](https://docs.docker.com/docker-for-mac/install/)
+1. Installing the global dependencies
+2. Installing the dependencies for IOS
+3. Installing the dependencies for Android
 
-💡 `$GOPATH` may need to be set explicitly (usually `$HOME/go`)
+<details><summary>Install the common dependencies (mandatory)</summary>
 
-### General React Native requirements
+1. Install `asdf` tool version manager following its [tutorial](https://asdf-vm.com/guide/getting-started.html).
 
-- Homebrew or package manager of choice
-- Node >= 14.x
-- The [yarn package manager](https://classic.yarnpkg.com/en/)
+> **Warning**: if you are using `nvm` you may face some incompatibility issues with `asdf` during the build process.
 
-### iOS dev requirements
+2. Run the initial setup script in the root project's folder:
+
+   ```bash
+   $ cd berty # root folder
+
+   $ make asdf.install_tools
+   ```
+
+   If you have an issue related with gpg, eg:
+   "Missing one or more of the following dependencies: tar, gpg"
+   You can try solve this issue using: `$ brew install gpg`
+
+3. Install the [General React Native requirements](#general-react-native-requirements)
+4. [Watchman](https://facebook.github.io/watchman/docs/install/) to enable live reloading
+5. [Docker Desktop](https://docs.docker.com/docker-for-mac/install/)
+6. `$GOPATH` may need to be set explicitly (usually `$HOME/go`)
```

Comment:
Doesn't asdf manage `$GOPATH`. The asdf Go home is not in `$HOME/go` .

Reply:
@clegirar could we remove line 99?

Reply:
Have you put something related to `$GOPATH` in your `.profile`/`.zshrc`/`.bashrc` ? If no, yeah you can delete it 👍